app/src/main/java/com/babylon/wallet/android/domain/usecases/AuthorizeSpecifiedPersonaUseCase.kt

```diff
@@ -40,23 +40,39 @@ class AuthorizeSpecifiedPersonaUseCase @Inject constructor(
 
     @Suppress("ReturnCount", "NestedBlockDepth", "LongMethod")
     suspend operator fun invoke(incomingRequest: IncomingRequest): Result<DAppData> {
-        var operationResult: Result<DAppData> = Result.failure(DappRequestFailure.InvalidRequest)
+        var operationResult: Result<DAppData> = Result.failure(DappRequestFailure.NotPossibleToAuthenticateAutomatically)
         (incomingRequest as? AuthorizedRequest)?.let { request ->
             (request.authRequest as? AuthorizedRequest.AuthRequest.UsePersonaRequest)?.let {
                 val authorizedDapp = dAppConnectionRepository.getAuthorizedDapp(
                     dAppDefinitionAddress = request.metadata.dAppDefinitionAddress
-                ) ?: return Result.failure(DappRequestFailure.InvalidRequest)
-
+                )
+                if (authorizedDapp == null) {
+                    respondWithInvalidPersona(incomingRequest)
+                    return Result.failure(DappRequestFailure.InvalidPersona)
+                }
                 val authorizedPersonaSimple = authorizedDapp
                     .referencesToAuthorizedPersonas
                     .firstOrNull { authorizedPersonaSimple ->
                         authorizedPersonaSimple.identityAddress ==
                             (request.authRequest as? AuthorizedRequest.AuthRequest.UsePersonaRequest)?.personaAddress
-                    } ?: return Result.failure(DappRequestException(DappRequestFailure.InvalidPersona))
+                    }
+                if (authorizedPersonaSimple == null) {
+                    respondWithInvalidPersona(incomingRequest)
+                    dAppMessenger.sendWalletInteractionResponseFailure(
```

Comment:
This is exactly what method above does. Is it duplicate ? 🤔

Reply:
not sure what you mean, could you explain? Which method you have in mind?

Reply:
`respondWithInvalidPersona` method has exactly the same content which is here :
```
dAppMessenger.sendWalletInteractionResponseFailure(
            incomingRequest.remoteConnectorId,
            incomingRequest.interactionId,
            WalletErrorType.InvalidPersona
        )
```
Right ? 
So you call the method respondWithInvalidPersona and below you paste exactly the same content again ?

Reply:
ahh yeah, ok updated